tests/integration/preprocessor/_supplementary_vars/test_add_supplementary_variables.py

```diff
@@ -281,6 +282,77 @@ def test_add_ancillary_vars_errors(self):
             np.ones(3),
             dim_coords_and_dims=[(plev_dim, 0)],
         )
-        msg = "No coordinate associated with ancillary"
-        with pytest.raises(ValueError, match=msg):
+        with pytest.raises(iris.exceptions.CoordinateNotFoundError):
             add_ancillary_variable(cube, ancillary_cube)
+
+    def test_get_data_dims(self):
+        """Test get_data_dims matching function."""
+        cube = iris.cube.Cube(
+            self.new_cube_3D_data,
+            dim_coords_and_dims=[
+                (self.depth, 0),
+                (self.lats, 1),
+                (self.lons, 2),
+            ],
+        )
+        plev_dim = iris.coords.DimCoord(
+            [0, 1.5, 3],
+            standard_name="air_pressure",
+            bounds=[[0, 1], [1, 2], [2, 3]],
+            units="Pa",
+        )
+        # Ancillary var has no dimension match
+        ancillary_dummy = iris.cube.Cube(
+            np.ones(3),
+            dim_coords_and_dims=[(plev_dim, 0)],
+        )
+        ancillary_dummy_cube = iris.coords.AncillaryVariable(
+            ancillary_dummy.core_data(),
+            standard_name=ancillary_dummy.standard_name,
+            units=ancillary_dummy.units,
+            var_name=ancillary_dummy.var_name,
+            attributes=ancillary_dummy.attributes,
+            long_name=ancillary_dummy.long_name,
+        )
+        with pytest.raises(iris.exceptions.CoordinateNotFoundError):
+            _ = get_data_dims(
+                cube,
+                ancillary_dummy,
+                ancillary_dummy_cube,
+            )
+        # Ancillary var has only one dimension match
+        ancillary_dummy = iris.cube.Cube(
+            np.ones((3, 3)),
+            dim_coords_and_dims=[(self.lats, 0), (plev_dim, 1)],
+        )
+        ancillary_dummy_cube = iris.coords.AncillaryVariable(
+            ancillary_dummy.core_data(),
+            standard_name=ancillary_dummy.standard_name,
+            units=ancillary_dummy.units,
+            var_name=ancillary_dummy.var_name,
+            attributes=ancillary_dummy.attributes,
+            long_name=ancillary_dummy.long_name,
+        )
+        with pytest.raises(iris.exceptions.CoordinateNotFoundError):
+            _ = get_data_dims(
+                cube,
+                ancillary_dummy,
+                ancillary_dummy_cube,
+            )
```

Comment:
This looks like a new test, please move it to a new `test_...` function (you can put the shared code in a [fixture](https://docs.pytest.org/en/6.2.x/fixture.html#back-to-fixtures)) or use a [parametrized test](https://docs.pytest.org/en/stable/how-to/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions) to generate separate tests for each scenario you are testing